test_schedule.py

```diff
@@ -91,16 +92,30 @@ def test_at_time(self):
         mock_job = make_mock_job()
         assert every().day.at('10:30').do(mock_job).next_run.hour == 10
         assert every().day.at('10:30').do(mock_job).next_run.minute == 30
+        assert every().day.at('10:30:50').do(mock_job).next_run.second == 50
 
     def test_at_time_hour(self):
         with mock_datetime(2010, 1, 6, 12, 20):
             mock_job = make_mock_job()
             assert every().hour.at(':30').do(mock_job).next_run.hour == 12
             assert every().hour.at(':30').do(mock_job).next_run.minute == 30
+            assert every().hour.at(':30').do(mock_job).next_run.second == 0
             assert every().hour.at(':10').do(mock_job).next_run.hour == 13
             assert every().hour.at(':10').do(mock_job).next_run.minute == 10
+            assert every().hour.at(':10').do(mock_job).next_run.second == 0
             assert every().hour.at(':00').do(mock_job).next_run.hour == 13
             assert every().hour.at(':00').do(mock_job).next_run.minute == 0
+            assert every().hour.at(':00').do(mock_job).next_run.second == 0
+
+    def test_at_time_minute(self):
+        with mock_datetime(2010, 1, 6, 12, 20, 30):
+            mock_job = make_mock_job()
+            assert every().minute.at(':40').do(mock_job).next_run.hour == 12
+            assert every().minute.at(':40').do(mock_job).next_run.minute == 20
+            assert every().minute.at(':40').do(mock_job).next_run.second == 40
+            assert every().minute.at(':10').do(mock_job).next_run.hour == 12
+            assert every().minute.at(':10').do(mock_job).next_run.minute == 21
+            assert every().minute.at(':10').do(mock_job).next_run.second == 10
```

Comment:
Let's add a couple of cases with bad inputs here (and to `test_at_time_hour()`) to ensure we're raising an exception  for stuff like `at('::2')`, `at('.2')`, `at('2')`, `at(2)` + whatever else cases you can come up with.

Reply:
`at('2:30:000001')`

Also I don't think we have test cases to ensure we're converting str -> int with base-10 (e.g. `07` doesn't get treated as octal/base 8). But that's a bonus ;-)

Reply:
Oh and I'm thinking anything that's not `HH:MM:SS`, `:MM` or `:SS` should raise an exception. Otherwise we'll have to support these malformed inputs forever (or break backwards compat in a future release)

Reply:
> Let's add a couple of cases with bad inputs here

Done.


> I don't think we have test cases to ensure we're converting str -> int with base-10

Would an existing test like the one below handle that?

`assert every().hour.at(':10').do(mock_job).next_run.minute == 10`

The idea being that if the `':10'` string was converted as octal, the next run's minute would not be the integer '10'.


> I'm thinking anything that's not `HH:MM:SS`, `:MM` or `:SS` should raise an exception. 

Done.  I assumed that `HH:MM` should also be a valid format despite not being in that list.

Reply:
Awesome!

> The idea being that if the ':10' string was converted as octal, the next run's minute would not be the integer '10'.

I was thinking we may want a test to confirm `07` gets parsed as base-10 `7` but maybe I'm being overly paranoid here.

Reply:
> I was thinking we may want a test to confirm `07` gets parsed as base-10 `7`

But the number '7' is the same in decimal and octal; the numbers start to change with '8'.  The octal value '10' would be equivalent to the decimal value '8'.

I think tests like the one below would fail if the strings were getting parsed as any numerical system other than decimal, because the value '10' on the right side is always going to be decimal:

`assert every().hour.at(':10').do(mock_job).next_run.minute == 10`